src/big/big_decimal.cr

```diff
@@ -239,16 +245,23 @@ struct BigDecimal
   end
 
   def to_i
-    puts self.to_s
-    to_s.to_i
+    if @value >= 0
+      (@value / TEN ** @scale)
+    else
+      -(@value.abs / TEN ** @scale)
+    end.to_i
```

Comment:
No, we never chain a method call after `end`. The correct way to write this is to stick the `to_i` on the end of the if branches, which works because of explicit return.

If this wasn't the final (only) statement in the method, we simply assign to a variable inside if:
```cr
if @value >= 0
  int_value = (@value / TEN ** @scale).to_i
else
  int_value = -(@value.abs / TEN ** @scale).to_i
end

# use int_value
```

Reply:
Righto. Fixed.